pandas/core/index.py

```diff
@@ -1677,33 +1690,52 @@ def get_indexer(self, target, method=None, limit=None):
 
         pself, ptarget = self._possibly_promote(target)
         if pself is not self or ptarget is not target:
-            return pself.get_indexer(ptarget, method=method, limit=limit)
+            return pself.get_indexer(ptarget, method=method, limit=limit,
+                                     tolerance=tolerance)
+
+        if tolerance is not None:
+            tolerance = self._convert_tolerance(tolerance)
```

Comment:
@jreback this is where I do the tolerance coercion... trying to remember why I moved it up here, I think this was to ensure it will be coerced properly before possibly converting the index to object dtype (on the lines below). I'll add a comment.